apps/hpc-ftsadmin/src/app/components/flow-form.tsx

```diff
@@ -0,0 +1,4094 @@
+                    {
+                      fileAssetID:
+                        uploadedFileArray[index] &&
+                        (uploadedFileArray[index] as { id?: number })?.id,
+                      reportFiles: [
+                        {
+                          fieldType: 'file',
+                          fileAssetID:
+                            uploadedFileArray[index] &&
+                            (uploadedFileArray[index] as { id?: number })?.id,
+                          type: 'file',
+                        },
+                      ],
+                      title: item.reportFileTitle,
+                      type: 'file',
+                    },
+                  ]
+                : [
+                    {
+                      title: item.reportUrlTitle,
+                      type: 'url',
+                      url: item.reportUrl,
+                    },
+                  ]
+              : uploadFileFlag
+              ? [
+                  {
+                    title: item.reportUrlTitle,
+                    type: 'url',
+                    url: item.reportUrl,
+                  },
+                ]
+              : item.reportFiles
+              ? item.reportFiles
+              : [],
+
+          reportChannel: {
+            group: 'reportChannel',
+            id: item.reportChannel && item.reportChannel.value,
+            name: item.reportChannel && item.reportChannel.displayLabel,
+          },
+        };
+      }),
+      flowType: {
+        id: values.flowType && values.flowType.value,
+        name: values.flowType && values.flowType.displayLabel,
+        group: 'flowType',
+      },
+      keywords: values.keywords.map((item) => ({
+        id: item.value,
+        name: item.displayLabel,
+        group: 'keywords',
+      })),
+      flowStatuses: {
+        id: values.flowStatus && values.flowStatus.value,
+        name: values.flowStatus && values.flowStatus.displayLabel,
+        group: 'flowStatus',
+      },
+      contributionTypes: {
+        id: values.contributionType && values.contributionType.value,
+        name: values.contributionType && values.contributionType.displayLabel,
+        group: 'contributionType',
+      },
+      method: {
+        id: values.method && values.method.value,
+        name: values.method && values.method.displayLabel,
+        group: 'method',
+      },
+      earmarking: values.earmarkingType
+        ? {
+            id: values.earmarkingType.value,
+            name: values.earmarkingType.displayLabel,
+            group: 'earmarkingType',
+          }
+        : null,
+      categories: [] as (string | number)[],
+      isCancellation: isPending ? true : !isPending ? false : null,
+      cancelled: isPending && isCancellation ? true : null,
+      pendingStatus: isPending ? true : !isPending ? false : [],
+      planEntities: isPending ? true : !isPending ? false : [],
+      planIndicated: isPending ? true : !isPending ? false : [],
```

Comment:
[Nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) would come handy in this case. Verify if it's the use case that you are expecting, but if I'm not wrong I think you made to check wether it's `undefined` or `null`.

```suggestion
      isCancellation: isPending ?? null,
      cancelled: isPending ?? null,
      pendingStatus: isPending ?? [],
      planEntities: isPending ?? [],
      planIndicated: isPending ?? [],
```